cmd/chihaya/config.go

```diff
@@ -94,6 +95,19 @@ func (cfg ConfigFile) CreateHooks() (preHooks, postHooks []middleware.Hook, err
 				return nil, nil, errors.New("invalid client approval middleware config: " + err.Error())
 			}
 			preHooks = append(preHooks, hook)
+		case "deniability":
+			var hCfg deniability.Config
+			err := yaml.Unmarshal(cfgBytes, &hCfg)
+			if err != nil {
+				return nil, nil, errors.New("invalid deniability middleware config: " + err.Error())
+			}
+			hook, err := deniability.New(hCfg)
+			if err != nil {
+				return nil, nil, errors.New("invalid deniability middleware config: " + err.Error())
+			}
+
+			// TODO this needs to run after the responseHook
```

Comment:
Yeah, this is a problem:
1. The response middleware overwrites everything it writes
2. This middleware needs to act differently if an infohash is not tracked, i.e. it needs to see the response immediately before we send it to the client.

We could make the response middleware its own "real" middleware and specify it in the config. Or we could split the `preHooks` section into `preResponsePreHooks` and `postResponsePreHooks` ?